src/DateTimePicker/PickerEnforcedInput.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { forwardRef } from 'react';
+
+import classNames from 'classnames';
+import { isValid } from 'date-fns';
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCalendarAlt } from '@fortawesome/pro-regular-svg-icons';
+
+type PickerEnforcedInputProps = {
+  disabled?: boolean;
+  inputClassName?: string;
+  name?: string;
+  startDate?: string;
+  value?: string;
+  onClick?: (...args: unknown[]) => unknown;
+};
+
+export const PickerEnforcedInput = forwardRef<HTMLElement, PickerEnforcedInputProps>(({
```

Comment:
So the crux of this error is: 
```
Types of property 'current' are incompatible.
        Type 'HTMLElement | null' is not assignable to type 'HTMLInputElement | null'.
```
`'HTMLElement | null' is not assignable to type 'HTMLInputElement | null'.

Using a more specific html element as the generic property fixes the issue.  If the forwarded ref were a form, for example you'd want to do like `forwardRef<HTMLFormElement,`
```suggestion
export const PickerEnforcedInput = forwardRef<HTMLInputElement, PickerEnforcedInputProps>(({
```